gokitmiddlewares/loggingmiddleware/middleware.go

```diff
@@ -47,6 +54,12 @@ func New(c Config) (endpoint.Middleware, error) {
 			}
 
 			defer func() {
+				var r interface{}
+				// Panics are handled as errors and re-raised
+				if r = recover(); r != nil {
+					log.Ctx(ctx).Warn().Msg("Logging endpoint middleware is handling an uncaught a panic. Please fix it!")
```

Comment:
Nit: this line could be swapped with the line below, so that `err` can be added to the warning message. This way, the panic message would in the same log as the warning and there would be no need to filter by request ID to know what has happened.